lhotse/workflows/whisper.py

```diff
@@ -39,32 +40,35 @@ def annotate_with_whisper(
 
     model = whisper.load_model(model_name, device=device)
 
-    for recording in recordings:
-        if recording.num_channels > 1:
+    for item in manifest:
+        if item.num_channels > 1:
             logging.warning(
-                f"Skipping recording '{recording.id}'. It has {recording.num_channels} channels, "
+                f"Skipping recording or cut '{item.id}'. It has {item.num_channels} channels, "
                 f"but we currently only support mono input."
             )
             continue
-        audio = torch.from_numpy(recording.resample(16000).load_audio()).squeeze(0)
+        audio = torch.from_numpy(item.resample(16000).load_audio()).squeeze(0)
         result = whisper.transcribe(model=model, audio=audio, language=language)
         supervisions = [
             SupervisionSegment(
-                id=f"{recording.id}-{segment['id']:06d}",
-                recording_id=recording.id,
+                id=f"{item.id}-{segment['id']:06d}",
+                recording_id=item.id
+                if isinstance(item, Recording)
+                else item.recording_id,
                 start=round(segment["start"], ndigits=8),
                 duration=round(segment["end"], ndigits=8),
                 text=segment["text"].strip(),
                 language=result["language"],
             )
             for segment in result["segments"]
         ]
-        cut = recording.to_cut()
+        cut = item.to_cut() if isinstance(item, Recording) else item
+        recording = item if isinstance(item, Recording) else item.recording
         if supervisions:
             supervisions = _postprocess_timestamps(supervisions)
             cut.supervisions = list(
                 trim_supervisions_to_recordings(
-                    recordings=recordings, supervisions=supervisions, verbose=False
+                    recordings=recording, supervisions=supervisions, verbose=False
```

Comment:
This won't do unfortunately. If you have 1h Recording with 10s cuts, the segments can go beyond cut's end (which doesn't make sense).

Unfortunately I think it might be cleaner to write a separate variant of this function for cuts... I don't like how the control flow became quite complex.